rustls/src/server/handy.rs

```diff
@@ -155,7 +158,8 @@ impl ResolvesServerCertUsingSni {
     pub fn add(&mut self, name: &str, ck: sign::CertifiedKey) -> Result<(), Error> {
         let checked_name = DnsNameRef::try_from(name)
             .map_err(|_| Error::General("Bad DNS name".into()))
-            .map(|dns| dns.to_lowercase_owned())?;
+            .map(|name| name.to_lowercase_owned())?;
+        let server_name = ServerName::DnsName(checked_name.clone());
```

Comment:
I think we should be able to avoid the `.clone()` here by reordering a bit?

Reply:
have managed this by moving the `DnsName` into and then out of the `ServerName` around the name verification, but better ideas welcome!